packages/core/src/EntityManager.ts

```
@@ -398,6 +398,34 @@ export class EntityManager<D extends IDatabaseDriver = IDatabaseDriver> {
     return [entities, count];
   }
 
+  /**
+   * TODO
+   */
+  async findByCursor<
+    Entity extends object,
+    Hint extends string = never,
+  >(entityName: EntityName<Entity>, where: FilterQuery<Entity>, options: FindOptions<Entity, Hint> = {}): Promise<Cursor<Entity, Hint>> {
+    const em = this.getContext(false);
+    entityName = Utils.className(entityName);
+    const meta = em.metadata.get<Entity>(entityName);
+
+    // ordering is required, default to order by PK
+    if (Utils.isEmpty(options.orderBy)) {
```

how about throw error instead of help user to choose PK as default?
because if PK is uuid, the result will be look weird.

---

I guess you are right, that's another good point with the uuid. I did this initially as I wanted to have defaults rather than throw, but then I realized the defaults (even if the value would be sequential) would have to work dynamically based on the cursor options probably, things started to look weird when I wrote the initial tests. 

Sidenote: maybe we should enforce the same for `em.findAndCount`?

---

Maybe we could use the PK default only if it's auto-increment?